### PR TITLE
faster reads

### PR DESCRIPTION
**Pull request by bfulton**
Avoid unnecessary locking on write cache or memory page flushing for reads.

@kbarrette @dstuebe

### CODE REVIEW

File: src/main/java/com/upserve/uppend/BlockedLongs.java [hunk 1]
```diff
@@ -147,9 +147,7 @@ public synchronized void append(final long pos, final long val) {
 
     public LongStream values(long pos) {
         log.trace("streaming values from {} at {}", file, pos);
-        if (outDirty.get()) {
```
Comment (kbarrette) [inline src/main/java/com/upserve/uppend/BlockedLongs.java]: Looks like these are the only two gets of `outDirty` - can it be removed?

File: src/main/java/com/upserve/uppend/AppendOnlyStore.java [hunk 2]
```diff
@@ -1,13 +1,51 @@
...
+     * Read byte arrays that have been stored under a given partition and key in
+     * the order they were stored, skipping the write cache of this store so
+     * that unflushed data is not visible
+     *
+     * @param partition the partition under which to retrieve
+     * @param key the key under which to retrieve
+     * @throws IllegalArgumentException if partition is invalid
+     * @return a stream of the stored byte arrays in storage order
+     */
+    Stream<byte[]> readSequentialFlushed(String partition, String key);
```
Comment (kbarrette) [inline src/main/java/com/upserve/uppend/AppendOnlyStore.java]: In what case would we not want to read in parallel?

File: src/main/java/com/upserve/uppend/AppendOnlyStore.java [hunk 3]
```diff
@@ -1,13 +1,51 @@
...
 
 import java.io.Flushable;
+import java.util.stream.Stream;
 
 /**
  * Add byte arrays under a key and partition, and retrieve them. Note the
  * expectation that the byte arrays are appended to the value, which is an
  * ever-growing list of byte arrays.
  */
 public interface AppendOnlyStore extends ReadOnlyAppendOnlyStore, Flushable {
```
Comment (kbarrette) [inline src/main/java/com/upserve/uppend/AppendOnlyStore.java]: Slightly concerned that what was once a very simple interface is growing

File: src/main/java/com/upserve/uppend/AppendOnlyStore.java (hunk 2 above)
Comment (dstuebe) [reply src/main/java/com/upserve/uppend/AppendOnlyStore.java]: So you can read in order

Comment (kbarrette) [reply src/main/java/com/upserve/uppend/AppendOnlyStore.java]: Understood, but what's the use case? When do we want to read in order?

File: src/main/java/com/upserve/uppend/BlockedLongs.java (hunk 1 above)
Comment (dstuebe) [reply src/main/java/com/upserve/uppend/BlockedLongs.java]: Good catch

**Pull request merged**